src/services/dynamic-choices-service.ts

```diff
@@ -45,13 +45,17 @@ export const DynamicChoicesService = {
     return selectedAppVersionId;
   },
 
-  async chooseAppAndAppVersion(options?: { appId?: number; useDefaultVersion?: boolean; useLiveVersion?: boolean }) {
-    const { appId, useDefaultVersion = false, useLiveVersion = false } = options || {};
-    const filterByStatus = useLiveVersion
-      ? [APP_VERSION_STATUS.LIVE, APP_VERSION_STATUS.DRAFT]
-      : [APP_VERSION_STATUS.DRAFT];
+  async chooseAppAndAppVersion(
+    useDeprecatedVersion: boolean,
+    useLiveVersion: boolean,
+    options?: { appId?: number; autoSelectVersion?: boolean },
```

Comment:
General comment:
Transfering `true, true` can cause issues as well.
What do you say about having an object for the arguments? **not optional**, but an object for better readability
```suggestion
  async chooseAppAndAppVersion(
    { useDeprecatedVersion: boolean,
    useLiveVersion: boolean },
    options?: { appId?: number; autoSelectVersion?: boolean },
```
This will make sure people understand what is **true** and what is **false**